tests/integration_tests/conftest.py

```diff
@@ -81,8 +82,46 @@ def test_db(scope="function"):
         drop_database(test_db_url)
 
 
+@pytest.fixture(scope="session")
+def data_db_connection() -> Generator[Connection, None, None]:
+    test_db_url = get_test_db_url()
+
+    if database_exists(test_db_url):
+        drop_database(test_db_url)
+    create_database(test_db_url)
+
+    saved_db_url = os.environ["DATABASE_URL"]
+    os.environ["DATABASE_URL"] = test_db_url
+
+    test_engine = create_engine(test_db_url)
+
+    run_migrations(test_engine)
+    connection = test_engine.connect()
+
+    yield connection
+    connection.close()
+
+    os.environ["DATABASE_URL"] = saved_db_url
+    drop_database(test_db_url)
+
+
+@pytest.fixture(scope="function")
+def data_db(data_db_connection):
+    transaction = data_db_connection.begin()
+
+    SessionLocal = sessionmaker(
+        autocommit=False, autoflush=False, bind=data_db_connection
+    )
+    session = SessionLocal()
+
+    yield session
+
+    session.close()
+    transaction.rollback()
+
+
 @pytest.fixture
-def data_db(scope="function"):
+def data_db_slow(scope="function"):
```

Comment:
What's the rationale for leaving this old fixture in?

Reply:
old commit